contributed_definitions/nyaml/NXpiezo_config_spm.yaml

```diff
@@ -0,0 +1,60 @@
+category: base
+doc: |
+  A base class to describe piezo settings for scanning probe microscopy.
+
+  The material of the Piezoelectric scanner is composed of polycrystalline solids and 
+  attached at the head of the tip. The material is highly sensitive to applied voltage. 
+# Discussion: No need to create this base class rather we can define in spm application definition.
+NXpiezo_config_spm(NXobject):
+  piezo_material(NXpiezoelectric_material):
+    doc: |
+      The material description and properties of the piezoelectric scanner materials.
+  calibration(NXcalibration):
+  
+    calibration_method(NXidentifier):
+      doc: |
+        The method used for calibration .
+      calibration_type(NX_CHAR):
```

Comment:
This should probably not be in `NXidentifier`, but one level above

Reply:
The `calibration_method(NXidentifier)` from `NXcalibration` base class. I just extended it for to attach `calibration_type` which is `SPM` specific field.